chalice/app.py

```diff
@@ -397,6 +414,30 @@ def json_body(self):
                     raise BadRequestError('Error Parsing JSON')
             return self._json_body
 
+    @property
+    def files(self):
+        content_type = self.headers.get('content-type')
+        if content_type.startswith('multipart'):
+            if self._files is None:
+                self._files = self._generate_files(
+                    content_type, self.raw_body
+                )
+            return self._files
+
+    def _generate_files(self, content_type, raw_body):
+        full_content = b'Content-Type: ' + \
+            content_type.encode('utf-8') + \
```

Comment:
This doesn't look like the right content type.  We're plumbing through the content type of the form data, not the content type of the file.

On second thought, why are we adding an explicit content type header?  That seems like the wrong thing to do.  If there's one available for the file we should use it.